app/src/main/java/io/appium/uiautomator2/utils/AXWindowHelpers.java

```diff
@@ -99,20 +104,43 @@ private static AccessibilityNodeInfo[] getWindowRoots() {
         return result.toArray(new AccessibilityNodeInfo[0]);
     }
 
+    private static AccessibilityNodeInfo getTopmostWindowRootFromActivePackage() {
+        CharSequence activeRootPackageName = getActiveWindowRoot().getPackageName();
+        List<AccessibilityWindowInfo> windows = getWindows();
+
+        Optional<AccessibilityNodeInfo> topmostWindowRootFromActivePackage = windows.stream()
+                .filter(window -> window.getRoot() != null)
+                .filter(window -> window.getRoot().getPackageName().toString().contentEquals(activeRootPackageName))
+                .max(Comparator.comparing(AccessibilityWindowInfo::getLayer))
+                .map(AccessibilityWindowInfo::getRoot);
+
+        return topmostWindowRootFromActivePackage
```

Comment:
I think we could merge this expression with the previous one into a single return

Reply:
Thanks. Addressed in 92279c4889d9dad61bc997aee83ae7cc2b299d15.